src/reducers/utils/fileUtils.ts

```diff
@@ -46,3 +46,12 @@ export function insert(children: any[], [head, ...tail]: any, code: string): IPr
     }
     return children;
 }
+
+// Insert item into tree structure overwriting items with same type and name
+export function appendWithoutDuplicate(baseArray: IProjectItem[], newArray: IProjectItem[]): IProjectItem[] {
```

Comment:
Sorting can be performed in this function, cause there would be no case when we need not sorted array.

Reply:
Moved sortProject to appendWithoutDuplicate